sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/CosmosMultiHashTest.java

```diff
@@ -217,22 +314,98 @@ private void validateDocCRUDandQuery() throws Exception {
             //Run the query.
             String query = String.format("SELECT * from c where c.id = '%s'", doc_current.get("id").asText());
 
-            CosmosPagedIterable<ObjectNode> feedResponseIterator1 =
+            CosmosPagedIterable<ObjectNode> feedResponseIterator =
                 createdMultiHashContainer.queryItems(query, queryRequestOptions, ObjectNode.class);
-            assertThat(feedResponseIterator1.iterator().hasNext()).isTrue();
+            assertThat(feedResponseIterator.iterator().hasNext()).isTrue();
 
             query = String.format("SELECT * from c where c.id = '%s'", doc_current.get("id").asText());
             queryRequestOptions = new CosmosQueryRequestOptions();
-            feedResponseIterator1 =
+            feedResponseIterator =
                 createdMultiHashContainer.queryItems(query, queryRequestOptions, ObjectNode.class);
-            assertThat(feedResponseIterator1.iterator().hasNext()).isTrue();
+            assertThat(feedResponseIterator.iterator().hasNext()).isTrue();
         }
 
-        String query = String.format("SELECT * from c where c.city = '%s'", docs.get(2).get("city").asText());
+        partitionKey = new PartitionKeyBuilder()
+            .add(doc5.get("city").asText())
+            .add(doc5.get("zipcode").asText())
+            .build();
+
+        String query = "SELECT * from c";
         CosmosQueryRequestOptions queryRequestOptions = new CosmosQueryRequestOptions();
-        CosmosPagedIterable<ObjectNode> feedResponseIterator1 =
+        queryRequestOptions.setPartitionKey(partitionKey);
+        CosmosPagedIterable<ObjectNode> feedResponseIterator =
             createdMultiHashContainer.queryItems(query, queryRequestOptions, ObjectNode.class);
-        assertThat(feedResponseIterator1.stream().count()).isEqualTo(3);
+        assertThat(feedResponseIterator.stream().count()).isEqualTo(2);
+
+        query = String.format("SELECT * from c where c.city = '%s'", docs.get(2).get("city").asText());
+        queryRequestOptions = new CosmosQueryRequestOptions();
+        feedResponseIterator = createdMultiHashContainer.queryItems(query, queryRequestOptions, ObjectNode.class);
+        assertThat(feedResponseIterator.stream().count()).isEqualTo(3);
 
+        query = String.format("SELECT * from c where c.city = '%s'", docs.get(0).get("city").asText());
+        partitionKey =
+            new PartitionKeyBuilder()
+                .add("Redmond")
+                .add("98053")
+                .build();
+        queryRequestOptions.setPartitionKey(partitionKey);
+        feedResponseIterator = createdMultiHashContainer.queryItems(query, queryRequestOptions, ObjectNode.class);
+        assertThat(feedResponseIterator.stream().count()).isEqualTo(2);
+
+        //Negative test - query option fails with incomplete partition key
+        //Not supported in query pipeline yet
```

Comment:
This feature (being able to use partial PK in queries) is the whole purpose of this feature. Why would we want to GA it without support for this scenario?

Reply:
@FabianMeiswinkel we have the same problems with the .NET SDK where when giving an incomplete PK will still return the query. The designs docs state that this scenario should fail. Currently this is not implemented in the backend but we were told that it should not be a blocker. Please verify this @SrinikhilReddy

Reply:
Update: As per the documentation "We support optimized query routing for partial specification of partition key paths in query filter as long the specified paths are the prefix in the paths definition i.e. if there are 3 partition keys in the paths the allowed values for partition key in header or in predicate are [ {path1} , {path1 and path2}, {path1, path2 and path3} ].  In each of these cases query is routed to a subset of physical partitions."

So the query in this case should succeed since the PK is `Redmond`, changing it to the `98053` should result in a failure.

Reply:
This is due to the work around with the EPK range resolution - working on those changes now.